code/game/machinery/computer/camera_console.dm

```diff
@@ -33,7 +34,6 @@
 
 /obj/structure/machinery/computer/cameras/Destroy()
 	SStgui.close_uis(src)
-	QDEL_NULL(current)
 	UnregisterSignal(src, COMSIG_CAMERA_MAPNAME_ASSIGNED)
```

Comment:
```suggestion
	current = null
	UnregisterSignal(src, COMSIG_CAMERA_MAPNAME_ASSIGNED)
```
You must ensure its nulled out then.

A good rule of thumb to know if you need to qdel something on destroy is if this thing is the creator of said thing; but past that things need to not hold on to references of other things else you hold up SSgarbage forcing a del() to eventually occur which will cause lag.